animate/metric.py

```diff
@@ -44,17 +44,11 @@ def __init__(self, function_space, *args, **kwargs):
             :class:`~.Function`. In this case, the function values are copied. If a
             :class:`~firedrake.mesh.MeshGeometry` is passed here then a tensor
             :math:`\mathbb P1` space is built on top of it.
+        :kwarg metric_parameters: same as for :func:`set_parameters <set_parameters>`.
         """
         if isinstance(function_space, fmesh.MeshGeometry):
             function_space = ffs.TensorFunctionSpace(function_space, "CG", 1)
-        self.metric_parameters = {}
-        self._variable_parameters = {
-            "dm_plex_metric_h_min": firedrake.Constant(1.0e-30),
-            "dm_plex_metric_h_max": firedrake.Constant(1.0e30),
-            "dm_plex_metric_a_max": firedrake.Constant(1.0e5),
-            "dm_plex_metric_boundary_tag": None,
-        }
-        self._variable_parameters_set = False
+        self.metric_parameters = kwargs.pop("metric_parameters", {})
```

Comment:
Please could you make this an explicit kwarg, rather than getting it from `**kwargs`?

Reply:
I tried that initially:
```
def __init__(self, function_space, metric_parameters={}, *args, **kwargs):
        # ...
        self.metric_parameters = metric_parameters
```
and everything else the same, but this results in 120 tests failing. I can't figure out for the life of me why. So I changed to `kwargs.pop` which I think is identical, and that works. Could you give it a try please?

Reply:
Hm yeah I see. Maybe it's something to do with `RiemannianMetric` being a subclass of `Function`. Not a major issue so let's just leave it for now.